test_lbaas.py

```diff
@@ -0,0 +1,625 @@
+        load_balancer.toggle_pool_member(pool, member_first, enabled=True)
+
+    # Assert the other backend is added back to the pool and starts to serve
+    # requests again
+    retry_for(seconds=10).or_fail(
+        load_balancer.verify_backend,
+        msg=f'Backend {backend_first.name} not added back to the pool '
+            f'within 10s.',
+        prober=prober,
+        backend=backend_first,
+    )
+
+    # Assert the persistent downloads are still active
+    assert prober.output_of(
+        'systemctl --user is-active wget-first') == 'active'
+    assert prober.output_of(
+        'systemctl --user is-active wget-second') == 'active'
+
+
+def test_private_frontend(create_server, image, create_load_balancer_scenario,
+                          private_network):
+    """ A load balancer can use a private network as it's frontend (VIP)
+    network to receive connections.
+
+    """
+
+    # Create a function scoped prober as we are going to attach this server
+    # the the private frontend network
+    prober = create_server(image=image['slug'], use_private_network=True)
+
+    # Add a subnet to the private frontend network
+    frontend_subnet = private_network.add_subnet(cidr='192.168.100.0/24')
+
+    # Attach the frontend network to the server used as an LB client
+    prober.update(
+        interfaces=[
+            {'network': 'public'},
+            {'network': prober.interfaces[1]['network']['uuid']},
+            {'network': private_network.info['uuid']},
+        ],
+    )
+
+    # Create a private load balancer setup with 1 backend on a private network
+    load_balancer, listener, pool, (backend, ), backend_network = \
+        create_load_balancer_scenario(
+            num_backends=1,
+            algorithm='least_connections',
+            frontend_subnet=frontend_subnet,
+            prober=prober,
+    )
+
+    # Assert the backend is reachable from the prober over the load balancer
+    load_balancer.verify_backend(prober, backend)
+
+
+def test_floating_ip(prober, create_load_balancer_scenario, floating_ip):
+    """ A floating IP can be assigned to a load balancer and used to receive
+    client connections.
+
+    """
+
+    # Create a load balancer setup with one backend on a private network
+    load_balancer, listener, pool, backends, private_network = \
+        create_load_balancer_scenario(num_backends=1)
+
+    # Assign floating IP to load balancer
+    floating_ip.assign(load_balancer=load_balancer)
+
+    # Assert load balancer is reachable on the floating IP after at most 20s
+    retry_for(seconds=20).or_fail(
+        prober.http_get,
+        msg='Load balancer not reachable on floating IP after 20s.',
+        url=construct_http_url(floating_ip.address),
+    )
+
+
+def test_floating_ip_reassign(prober, create_load_balancer_scenario,
+                              floating_ipv4, server):
+    """ Test if a floating IP can be reassigned from a server to a load
```

Comment:
We generally write `Floating IP`, as it's a product name.

Reply:
fixed